### PR TITLE
Fix for the wrong callback

### PR DESCRIPTION
**Pull request by ashinagawa**
This commit fixes the issue described below.

With MoPub integration, at very first ad request to Vungle, adapter will send “adLoadFailed” callback since SDK is not initialized yet at this timing. The router class tries to call loadPlacementWithID while initializing SDK and it’ll fail.
This will nil out MoPub “rewardedAd” instance (since we are sending ‘loadAdFail’ callback) and a pub can’t play the ad even though SDK will send ‘adLoaded’ callback later after initialization done.

### CODE REVIEW

File: Vungle/VungleRouter.m [hunk 1]
```diff
@@ -139,7 +139,9 @@ - (void)requestAdWithCustomEventInfo:(NSDictionary *)info delegate:(id<VungleRou
         if (error) {
             MPLogInfo(@"Vungle: Unable to load an ad for Placement ID :%@, Error %@", placementId, error);
         }
-        [delegate vungleAdDidFailToLoad:error];        
+        if (self.sdkInitializeState == SDKInitializeStateInitialized) {
```
Comment (dravyan) [inline Vungle/VungleRouter.m]: @ashinagawa , can you help to clarify why the Vungle load error is sent when SDK is initialized? Also here`vungleAdDidFailToLoad` is to notify MoPub, in that case, you do not need to check whether Vungle SDK is initialized.

Comment (ashinagawa) [reply Vungle/VungleRouter.m]: @dravyan That was the exactly the issue which the pub faced.
While Vungle SDK initializing, if an ad is requested, Vungle SDK returns `vungleAdDidFailToLoad:error` as you see in "old" code. But, after initialization is done, Vungle SDK will request an ad and return `vungleAdDidLoad`. At this point, MoPub SDK already nil-out "custom event instance" becasue of the previous `vungleAdDidFailToLoad:error` callback. So, when a publisher try to play the ad, there is no "custom event instance"  anymore. 

So, out fix is "not to send `vungleAdDidFailToLoad:error` at this point.

Comment (dravyan) [reply Vungle/VungleRouter.m]: @ashinagawa , feel free to close this PR or update the same when the updated fix ready. Thanks!

Comment (dravyan) [reply Vungle/VungleRouter.m]: @ashinagawa , please let us know if we are good to close this PR?

Comment (ashinagawa) [reply Vungle/VungleRouter.m]: @dravyan Sorry for late reply Yes. I will close this for now.

**Pull request closed without merging**